contrib/pylightning/lightning/plugin.py

```diff
@@ -306,9 +306,20 @@ def _getmanifest(self):
                 doc = "Undocumented RPC method from a plugin."
             doc = re.sub('\n+', ' ', doc)
 
+            argspec = inspect.getargspec(func)
+            args = argspec.args[1:]
```

Comment:
My assumption here was that `plugin` had to be the first arg, but @cdecker pointed out that `request` or `plugin` can be _any_ positional arg.

This need to be updated to
```
args = [arg for arg in argspec.args if arg not in ('plugin', 'request')]
```

One confusion here is that [this comment](https://github.com/ElementsProject/lightning/blob/master/contrib/pylightning/lightning/plugin.py#L64-L66)  seems to suggest `request` and `plugin` should be the last args, but that does not seem to actually matter. Maybe I am reading this wrong.

Reply:
Apologies for turning this commit into a :hankey: This is what you _really_ want
```python
            argspec = inspect.getargspec(func)
            defaults = argspec.defaults
            num_defaults = len(defaults) if defaults else 0
            args = []
            for idx, arg in enumerate(argspec.args):
                if arg in ('plugin', 'request'):
                    continue
                # Positional arg
                if idx < (num_defaults):
                    args.append("%s" % arg)
                # Keyword arg
                else:
                    args.append("[%s]" % arg)
```

Reply:
Applied!